common/logger/src/filter.rs

```diff
@@ -0,0 +1,348 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{Level, Metadata};
+use std::{env, str::FromStr};
+
+pub struct FilterParseError;
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum LevelFilter {
+    Off,
```

Comment:
Perhaps, this was supported in the old filtering (provided by env-logger) and i think theres value in continuing to support it. It needs to be distinct from Level because there is no `Off` level that can be generated by a log event. This way we're keeping these two concepts, filtering and event level different so as to ensure at compile time we can't get into a bad state (ie its unrepresentable)